lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +154,36 @@ def self.update_command
     'update'
   end
 
+  def get_wanted_version(should)
+    if should =~ /[<>=~]/
```

Comment:
after chatting we concluded that we shouldn't rely on specific characters here, but on the String class itself
```suggestion
    if should.is_a?(String)
```